torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -1250,6 +1250,8 @@ def call_function(
         if "context_fn" in kwargs and kwargs["context_fn"] != noop_context_fn:
             context_fn = kwargs.pop("context_fn")
             self.value.context_fn = context_fn.fn
+        else:
+            self.value.context_fn = None
```

Comment:
To check my understanding,

What is happening here is `self.value` in this case is the following global object
https://github.com/pytorch/pytorch/blob/16cb3bdd57d00ccd4e658431b88b6acea7f3df6e/torch/_higher_order_ops/wrap.py#L159

And `context_fn` is an attribute on that global object.
https://github.com/pytorch/pytorch/blob/16cb3bdd57d00ccd4e658431b88b6acea7f3df6e/torch/_higher_order_ops/wrap.py#L87

 So, while testing if we run the complete suit, this fails because a previous test sets `context_fn` without patching the experimental flag.

I have verified that running a failing test in isolation leads to the test passing because that state is ok then (e.g. `pytest test/dynamo/test_activation_checkpointing.py -v -k test_kwargs -s`)

I am curious as to why this passes in CI?

cc: @zou3519 @anijain2305

Reply:
I do set the experimental flag, and it fails.

Seems like it is flaky. e.g. https://github.com/pytorch/pytorch/issues/109862

Reply:
We should not be modifying the `tag_activation_checkpoint` singleton, so a better solution would be that we stop doing that, right?

Reply:
> We should not be modifying the tag_activation_checkpoint singleton, so a better solution would be that we stop doing that, right?

This may seem reasonable, but actually, the express purpose of `ctx_function` is that it is modified by higher order op to propagate a setting when tracing with dynamo.

Reply:
Could we merge this. These tests have been flakey without this simple fix. https://github.com/pytorch/pytorch/issues/112678 

CC @Chillee